src/jit.rs

```diff
@@ -1300,7 +1300,7 @@ impl JitCompiler {
                         // Workaround for unresolved symbols in ELF: Report error at runtime instead of compiletime
                         emit_rust_call(self, Executable::<E, I>::report_unresolved_symbol as *const _, &[
                             Argument { index: 2, value: Value::Constant64(self.pc as i64, false) },
-                            Argument { index: 1, value: Value::Constant64(executable as *const _ as i64, false) },
+                            Argument { index: 1, value: Value::Constant64(executable.as_ref() as *const _ as i64, false) },
```

Comment:
This is the critical change.
See: https://github.com/solana-labs/rbpf/pull/231/files#diff-7d39efd566d08b201d1e747f11b3909be0d39fd3d4c89cfd28e0b21bab3ec27aL1302